### PR TITLE
46 2d azimuth nelevation 2 position

### PR DESCRIPTION
**Pull request by Tim-Behrer**
Added a camera class, made simple changes to readme and tried to start of ahmed's simplified model

### CODE REVIEW

File: RGV Path Sim/Sensors/nl_leastSquares.m
Comment (katebenic) [inline RGV Path Sim/Sensors/nl_leastSquares.m:1]: This function is incomplete and should be removed from the commit prior to merging with main

Comment (LyonFoster) [reply RGV Path Sim/Sensors/nl_leastSquares.m:1]: I think it's fine to have incomplete stuff as long as there's a corresponding issue that Tim assigned to himself

**Comment by LyonFoster** [comment]
I think we should just try to get @aidanwoodcu 's in before merging this one in otherwise we could end up with a git-style bruh moment

File: Coordinate Frame Transitions/rotMat321.m [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+% % % attitude = 3 x 1 vector with the 3-2-1 Euler angles of the camera in the form of relative attitude
+% % %          = [theta_1_cam2uasBody,theta_2_cam2uasBody,theta_3_cam2uasBody]
+% Output: 
+% % % rotMat321 = the 3-2-1 rotation matrix parameterized by those angles.
+% Methodology: This function converts the camera coordinate frame to the uas body camera frame using a 321 rotation matrix
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
+
+%% Create the Rotation Matrix 
+% 321 Rotation Matrix 
+rotMat = [cos(attitude(2))*cos(attitude(3)), cos(attitude(2))*sin(attitude(3)), -sin(attitude(2)); ...
```
Comment (LyonFoster) [inline Coordinate Frame Transitions/rotMat321.m:16]: `dcm = angle2dcm( yaw, pitch, roll )` would this be simpler?

File: RGV Path Sim/Sensors/nl_leastSquares.m [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+function vec_uas2obj_in_uasBody = nl_leastSquares(vec_uasState,vec_azimuth_in_cam)
+%% Create a function that takes in an applicable vecor of azimuth measurements and ....
+% derives an expected position using non-linear least squares method
+% This function should effectively act as a low-level course localization
+% estimator
+
+%%% Current assumptions - the target being measured is not moving
+%%%                     - the camera azimuth determined is perfect
+%%%                     - the uasState vector is perfect
+%TODO(TB) - encorperate a constant moving target
```
Comment (LyonFoster) [inline RGV Path Sim/Sensors/nl_leastSquares.m:14]: make sure these get corresponding issues

File: Coordinate Frame Transitions/rotMat321.m (hunk 1 above)
Comment (Tim-Behrer) [reply Coordinate Frame Transitions/rotMat321.m:16]: I have just had issues with that function in the past, so I usually prefer to type it out manually

**Comment by Tim-Behrer** [comment]
> I think we should just try to get @aidanwoodcu 's in before merging this one in otherwise we could end up with a git-style bruh moment

I can change the PR to merging with main, id like the readME and the cam class to be common resourse soon. not as worried about the other functions

**Pull request closed without merging**